include/strbuf.h

```diff
@@ -0,0 +1,43 @@
+#ifndef POKEPLATINUM_STRBUF_H
+#define POKEPLATINUM_STRBUF_H
+
+typedef struct Strbuf_t Strbuf;
+typedef u16 Charcode;
+
+/*
+ * Control codes for how to pad numeric strings.
+ *
+ * Ex: `1000000` being assigned to a buffer of size 20 may be:
+ * - Left-justified:                      "1000000             "
+ * - Right-justified, padded with spaces: "             1000000"
+ * - Right-justified, padded with zeroes: "00000000000001000000"
+ */
+typedef enum {
+    PADDING_MODE_NONE,
+    PADDING_MODE_SPACES,
+    PADDING_MODE_ZEROES,
+} PaddingMode;
+
+Strbuf* Strbuf_Init(u32 size, u32 heapID);
+void Strbuf_Free(Strbuf *strbuf);
+void Strbuf_Clear(Strbuf *strbuf);
+void Strbuf_Copy(Strbuf *dst, const Strbuf *src);
+Strbuf* Strbuf_Clone(const Strbuf *src, u32 heapID);
+void Strbuf_FormatInt(Strbuf *dst, int num, u32 maxDigits, PaddingMode mode, BOOL whichCharset);
+void Strbuf_FormatU64(Strbuf *dst, u64 num, u32 maxDigits, PaddingMode mode, int whichCharset);
```

Comment:
Double check whether whichCharset is BOOL.

Reply:
Oh, now I see what you meant. Yeah, they should be the same type on either function.